File: SubsonicSwift/Models/Album.swift
import Foundation

struct Album: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let artist: String?
    let artistId: String?
    let coverArt: String?
    let songCount: Int?
    let year: Int?
    let genre: String?
    let duration: Int?
    var songs: [Song]?

    enum CodingKeys: String, CodingKey {
        case id, name, artist, artistId, coverArt, songCount, year, genre, duration
        case songs = "song"
    }

    static func == (lhs: Album, rhs: Album) -> Bool {
        lhs.id == rhs.id
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
}

struct AlbumResponse: Codable {
    let album: Album
}

struct AlbumListResponse: Codable {
    let albumList2: AlbumList

    struct AlbumList: Codable {
        let album: [Album]?
    }
}

File: SmashSonic-Android/app/src/main/java/com/smashsonic/data/model/Album.kt
package com.smashsonic.data.model

data class Album(
    val id: String,
    val name: String,
    val artist: String? = null,
    val artistId: String? = null,
    val coverArt: String? = null,
    val songCount: Int? = null,
    val year: Int? = null,
    val genre: String? = null,
    val duration: Int? = null,
    val songs: List<Song>? = null,
)
